cmd/neofs-node/cache.go

```diff
@@ -325,3 +327,37 @@ func (f *cachedIRFetcher) InnerRingKeys() ([][]byte, error) {
 
 	return val.([][]byte), nil
 }
+
+type ttlMaxObjectSizeCache struct {
+	mtx         sync.RWMutex
+	lastUpdated time.Time
+	lastSize    uint64
+	src         putsvc.MaxSizeSource
+}
+
+func newCachedMaxObjectSizeSource(src putsvc.MaxSizeSource) putsvc.MaxSizeSource {
+	return &ttlMaxObjectSizeCache{
+		src: src,
+	}
+}
+
+func (c *ttlMaxObjectSizeCache) MaxObjectSize() uint64 {
+	const ttl = time.Second * 30
+
+	c.mtx.RLock()
+	valid := time.Since(c.lastUpdated) < ttl
+	lastSize := c.lastSize
+	c.mtx.RUnlock()
+
+	if valid {
+		return lastSize
+	}
+
+	c.mtx.Lock()
+	size := c.src.MaxObjectSize()
```

Comment:
At this point other routine could already refresh cached value. Maybe we should re-check the cache to prevent redundant request?

Reply:
do you mean a network update after only one `if valid` check?

Reply:
Fixed.